v2/esp-entity-table.ts

```diff
@@ -59,7 +59,8 @@ export class EntityTable extends LitElement {
         } as entityConfig;
         this.entities.push(entity);
         this.entities.sort((a, b) => (a.name < b.name ? -1 : 1));
-        this.has_controls ||= this.control(entity).length||0>0
+        this.has_controls ||=
+          JSON.stringify(this.control(entity)) !== JSON.stringify(html``);
```

Comment:
I'm no expert in JavaScript / Lit but I think the better solution would be to change the line below to `return nothing;` (don't forget to add `nothing` to the import on line 1).

https://github.com/esphome/esphome-webserver/blob/7747f309ce9a28ad216a988c3faae9bb4172d361/v2/esp-entity-table.ts#L273 

After that you can simply test if the control equals nothing.

`this.has_controls ||= this.control(entity) == nothing;`

Reply:
Can't do that because it expects a template result type sadly. Tried this initially.